target_chains/ethereum/contracts/contracts/entropy/EntropyUpgradable.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@pythnetwork/entropy-sdk-solidity/EntropyErrors.sol";
+
+import "./EntropyGovernance.sol";
+import "./Entropy.sol";
+
+contract EntropyUpgradable is
+    Initializable,
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    Entropy,
+    EntropyGovernance
+{
+    event ContractUpgraded(
+        address oldImplementation,
+        address newImplementation
+    );
+
+    // The contract will have an owner and an admin
+    // The owner will have all the power over it.
+    // The admin can set some config parameters only.
+    function initialize(
+        address owner,
+        address admin,
+        uint pythFeeInWei,
+        address defaultProvider
+    ) public initializer {
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+
+        Entropy._initialize(admin, pythFeeInWei, defaultProvider);
+
+        // We need to transfer the ownership from deployer to the new owner
+        transferOwnership(owner);
```

Comment:
note for the future:

we should add a `safeTransferOwnership` function that has two stages:
- proposeOwnershipTransfer(newOwner)
- acceptOwnershipTransfer()

where the second one checks that msg.sender is the newOwner and calls transferOwnership. This approach guarantees that the new owner can send transactions, so we don't accidentally transfer to the wrong address.

This can be done in a separate PR though.